docs/src/references/tedge-configuration-management.md

```diff
@@ -0,0 +1,365 @@
+```
+
+```run command="tedge-configuration-plugin --help" lang="text" title="Output"
+Thin-edge device configuration manager
+
+USAGE:
+    tedge-configuration-plugin [OPTIONS]
+
+OPTIONS:
+        --config-dir <CONFIG_DIR>
+            [default: /etc/tedge]
+
+        --debug
+            Turn-on the debug log level.
+
+            If off only reports ERROR, WARN, and INFO If on also reports DEBUG and TRACE
+
+    -h, --help
+            Print help information
+
+    -V, --version
+            Print version information
+
+On start, `tedge-configuration-plugin` notifies of its managed configuration files and
+sends this list via MQTT.
+`tedge-configuration-plugin` subscribes to the topics for the commands `config_snapshot`
+and `config_update`.
+
+The thin-edge `CONFIG_DIR` is used:
+  * to find the `tedge.toml` where the following configs are defined:
+    ** `mqtt.bind.address` and `mqtt.bind.port` to connect to the tedge MQTT broker
+    ** `root.topic` and `device.topic`: for the MQTT topics to publish to and subscribe from
+  * to find/store the `tedge-configuration-plugin.toml`: the plugin configuration file
+```
+
+## Logging
+
+The `tedge-configuration-plugin` reports progress and errors on its `stderr`.
+
+* All upload and download operation commands are logged, when received and when completed,
+  with one line per file.
+* All changes to the list of managed file is logged, one line per change.
+* All errors are reported with the operation context (upload or download? which file?).
+
+## Open questions/opinions
+
+* Publishing the supported types to both `~/cmd/config_snapshot` and `~/cmd/config_update`?
+Mapper will decide how to address them? e.g. using the supported types from only `~/cmd/config_snapshot` and ignore the ones from `~/cmd/config_update`.
+* Logging. Are we now creating a file per config operation as described in the c8y-configuration-plugin spec?
+Not just logging to journalctl? Need to check.
+* Notification mechanism. The c8y-configuration-plugin publishes a notification message on a change in main device, in order that the 3rd components can detect the change.
+For example, mosquitto must be restarted once mosquitto.conf is changed.
+However, do we still need this mechanism?
+I think the successful command status update can be a replacement.
+A component can subscribe to `te/device/main///cmd/config_update/+`, then filer `"status": "successful"` and `"type": "mosquitto"`?
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main///cmd/config_update/1234' '{
+  "status": "successful",
+  "tedgeUrl": "http://127.0.0.1:8000/tedge/file-transfer/main/config_update/mosquitto-1234",
+  "type": "mosquitto",
+  "remoteUrl": "http://www.my.url",
+  "path": "/etc/mosquitto/mosquitto.conf"
+}'
+```
+
+* File deletion.
+  * Snapshot (upload): Mapper should delete the file after uploading to cloud.
+  * Update (download): It's possible that the plugin sends a DELETE request after downloading a file completed.
```

Comment:
The plugin shouldn't be doing that. The mapper created the file in the file transfer repository. So, it owns that file and hence should be responsible for any cleanup as well. Not just the file, but the symlinks as well.